src/journal/compress.c

```diff
@@ -41,6 +41,61 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_decompressionContext_t, LZ4F_freeDecompressionC
 #if HAVE_ZSTD
 DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_CCtx *, ZSTD_freeCCtx);
 DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_DCtx *, ZSTD_freeDCtx);
```

Comment:
hmm, what are the implications if we don#t add this compat kludge? how many and which CIs do we lose if we do?

Reply:
Your pick, cutting out the kludge is easy

Reply:
@evverx @mrc0mmand what do you say, how valuable is it to keep this kludge in? I'd normally just require a new enough library version, but if it helps bigly for the CIs we can keep the kludge in. You decide.

Reply:
My concern was, that the library is not in debian stable (but its available in the backports repo).
Its an extra commit that can be reverted at any time. I will rework the commits a bit, so that the first depends on zstd >= 1.4.0 and the commit containing the `kludge` pulling this down to 1.3.1.

Reply:
I don't think '1.3.*' should be supported so I'm inclined to say that the commit should be simply dropped. @ddstreet I  wonder if it would be possible to put the latest version of `libzstd-dev` into https://launchpad.net/~upstream-systemd-ci/+archive/ubuntu/systemd-ci?

On Semaphore CI Debian Buster is used but it can be fixed by editing "semaphoreci/semaphore-runner.sh". @nolange could you pull the backports repository there before installing `libzstd-dev` and `zstd`?

Reply:
> I don't think '1.3.*' should be supported so I'm inclined to say that the commit should be simply dropped. @ddstreet I wonder if it would be possible to put the latest version of `libzstd-dev` into https://launchpad.net/~upstream-systemd-ci/+archive/ubuntu/systemd-ci?

yep, libzstd 1.4.4 is in the ppa now and will be available to Ubuntu CI.

I can change the build/test dependency also to require libzstd-dev/zstd >= 1.4.0, if the code does require that.

Reply:
> I don't think '1.3.*' should be supported so I'm inclined to say that the commit should be simply dropped. @ddstreet I wonder if it would be possible to put the latest version of `libzstd-dev` into https://launchpad.net/~upstream-systemd-ci/+archive/ubuntu/systemd-ci?
> 
> On Semaphore CI Debian Buster is used but it can be fixed by editing "semaphoreci/semaphore-runner.sh". @nolange could you pull the backports repository there before installing `libzstd-dev` and `zstd`?

Actually no, unless you tell me exactly what line I should put where. Backports would be `deb http://deb.debian.org/debian buster-backports main`, not some ubuntu ppas?